src/core/killall.c

```diff
@@ -177,8 +178,11 @@ static int killall(int sig, Set *pids, bool send_sighup) {
                 }
 
                 if (kill(pid, sig) >= 0) {
-                        if (pids)
-                                set_put(pids, ULONG_TO_PTR(pid));
+                        if (pids) {
+                                r = set_put(pids, ULONG_TO_PTR(pid));
+                                if (r < 0)
+                                        log_oom();
```

Comment:
We could do that, but are we really not interested in logging this? If set_put fails here that pid will not be included in the list that we will wait for. That might lead to some unexpected behavior for that pid and with no explanation in the logs at all.